ext/rbtrace.c

```diff
@@ -803,17 +804,37 @@ rbtracer_watch(uint32_t threshold, bool cpu_time)
   }
 }
 
+static int
+get_msgq_key(pid_t pid)
+{
+#ifndef __linux__
+  return(pid);
+#else
+  char buffer[128];
+  int status;
+  struct stat buf;
+  snprintf(buffer, 128, "/proc/%d", pid);
```

Comment:
The largest pathname possible here is `6 + 10 + 1 == 17`. 128 is slight overkill for the buffer length 😄